src/imputors/drop.jl

```diff
@@ -36,38 +46,134 @@ NOTES (or premature optimizations):
     3. reshaping the data back to the desired shape.
 
 # Arguments
-* `imp::Drop`: this `Imputor` method
-* `ctx::Context`: contextual information for missing data
+* `imp::DropObs`: this `Imputor` method
 * `data::AbstractMatrix`: the data to impute
 
 # Returns
 * `AbstractMatrix`: a new matrix with missing rows removed
 """
-function impute!(imp::Drop, ctx::Context, data::AbstractMatrix)
-    ctx.num = size(data, 1)
-    mask = map(i -> !ismissing(ctx, data[i, :]), 1:size(data, 1))
-    return data[mask, :]
+function impute!(imp::DropObs, data::AbstractMatrix)
+    imp.context() do c
+        mask = map(i -> !ismissing(c, data[i, :]), 1:size(data, 1))
```

Comment:
```suggestion
        mask = map(i -> !ismissing(c, @view(data[i, :])), axes(data, 1))
```
Could even `@inbounds` it if super keen

another option:

```suggestion
        mask = vec(mapslices(col -> !ismissing(c, col), data; dims=1))
```

Reply:
I think I prefer the first option, but I might breakup the `map` across multiple lines and use the `view` function.